### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-radio-button

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1484]
## Description
This task describes the Storybook implementation for the sd-radio-button component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-radio-button can be found here: [sd-radio-button Notes](https://www.notion.so/matuzo/sd-radio-button-10cf3f6e1bc0805b9963dc03fe6e6863).

---------------------------------

## Comment
Problem: The biggest problem with this pattern is the mismatch between the actions people expect to take and those they actually can. If something looks like a button, you expect that you can use it like a button. For example, press Tab to focus it, but that only works for the selected option here. Please read Leonie Watson's article in the additional resources for more detail. I'm not necessarily saying that you must take action, but I'd consider it at least.
[New ticket](https://github.com/solid-design-system/solid/issues/1587) created to address this issue.

Problem: The examples with icons only have no accessible name.
Solution: Provide a label for the icons.

## Notes
- Disabled radio buttons: See [@sd-button](https://www.notion.so/matuzo/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=25). Everything I wrote there about disabling form elements applies here as well.
 [ ] Will be discussed with the disabled buttons

- All the buttons in the demos that are not part of a group of buttons are not focusable (`tabindex=-1`)
 [ ] Add the radio-group to the default story 

## Best Practices
- Prefer real radio buttons over button radio buttons if possible.

## Additional Resources
- https://tink.uk/perceived-affordances-and-the-functionality-mismatch/

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
